Use versioned target population

patriciagarcia opened the pull request:
Connects #20 

- Updates the service to use the new versioned target population field
- Makes sure that if no version is provided for any of the factors, version 1 is used by default

patriciagarcia committed:
refactor: extract factor calculation to own function

patriciagarcia committed:
feat: use the new versioned targetPopulation

BREAKING_CHANGE: requires location docs to contain an array of
target population versions

patriciagarcia committed:
feat: default all factors to version 1

patriciagarcia commented:
Semantic release failing, it was working for me locally but I made the mistake of removing my `node_modules` folder to be able to replicate the error. Now I can see the error but I have no idea how to fix it ;-). 

Any clue https://travis-ci.org/fielded/angular-nav-thresholds?

patriciagarcia commented:
Uh, actually what fails is `rollup -c` and it creates a weir file called `-c` that contains `undefined`

```
> npm run -s build && karma start
resolve failed:  { [Error: Cannot find module 'babel-runtime'] code: 'MODULE_NOT_FOUND' } babel-runtime
resolve failed:  { [Error: Cannot find module '@semantic-release'] code: 'MODULE_NOT_FOUND' } @semantic-release
Aborting commit due to empty commit message.
```

tlvince commented:
Pinning didn't work, but can reproduce locally.

tlvince commented:
Error is due to some publishing mess in https://github.com/rollup/babel-preset-es2015-rollup. Either way, that package is deprecated. Will push a fix direct to the branch here.

tlvince committed:
chore: use vanilla babel preset

babel-preset-es2015-rollup has been deprecated as you can now configure the
vanilla preset to exclude modules (which rollup handles).

patriciagarcia committed:
fix: update location target population field name

patriciagarcia committed:
fix: stay backwards compatible with old location docs

tlvince commented on src/thresholds.service.js:
```diff
@@ -40,40 +40,73 @@ const getFactor = (location, versions, version) => {
-    return {}
-  }
+  // centralized for whenever we implement #16
+  const somethingIsWrong = () => undefined
```
😄

patriciagarcia replied:
😂  so descriptive!

patriciagarcia commented:
@tlvince, updated to take into account the proper target population field naming and to make it backwards compatible with non versioned TPs
